client/app/lib/components/form/Form.tsx

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Button, Slide, Typography } from '@mui/material';
+import { ReactNode } from 'react';
+import useEmitterFactory, { Emits } from 'react-emitter-factory';
+import { Control, FormState, useForm, UseFormWatch } from 'react-hook-form';
+import { AnyObjectSchema } from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import useTranslation from 'lib/hooks/useTranslation';
+import translations from 'lib/translations/form';
+
+export interface FormEmitter {
+  reset?: () => void;
+  resetTo?: (data) => void;
+  setError?: (fieldName: string, errors: Record<string, string>) => void;
+}
+
+interface FormProps extends Emits<FormEmitter> {
+  initialValues?: Record<string, any>;
+  onSubmit?: (data) => void;
+  headsUp?: boolean;
+  validates?: AnyObjectSchema;
+  children?: (
+    control: Control,
+    watch: UseFormWatch<any>,
+    formState: FormState<any>,
+  ) => ReactNode;
+}
+
+const Form = (props: FormProps): JSX.Element => {
+  const { t } = useTranslation();
+  const { control, formState, reset, watch, handleSubmit, setError } = useForm({
+    defaultValues: props.initialValues,
+    resolver: props.validates && yupResolver(props.validates),
+  });
+
+  useEmitterFactory(props, { reset: () => reset(), resetTo: reset, setError });
+
+  return (
+    <form onSubmit={props.onSubmit && handleSubmit(props.onSubmit)}>
+      {props.children?.(control, watch, formState)}
+
+      {props.headsUp && (
+        <Slide in={formState.isDirty} direction="up" unmountOnExit>
+          <div className="fixed inset-x-0 bottom-0 z-10 flex w-full items-center justify-between bg-neutral-800 px-8 py-4 text-white sm:bottom-8 sm:mx-auto sm:w-fit sm:rounded-lg sm:drop-shadow-xl">
+            <Typography variant="body1">
+              {t(translations.unsavedChanges)}
+            </Typography>
+
+            <div className="ml-10">
+              <Button onClick={(): void => reset()} className="mr-4">
+                {t(translations.reset)}
+              </Button>
+
+              <Button type="submit" variant="contained" disableElevation>
```

Comment:
I tried using `LoadingButton` before, but the button and loading indicator cannot be seen on dark backgrounds. This is because MUI styles disabled components as having neutrals _with lower opacity_, as opposed to assigning them to the true neutrals. So, on dark backgrounds, the disabled buttons got darker instead of being lighter.

My 'hack' by manually setting the background colour worked, but not ideal. We should re-investigate this when we consolidate our theme; see how to handle disabled colours and colours on dark backgrounds/themes.